### DISCUSSION TITLE
Managing image versions in Testcontainers modules

### DISCUSSION THREAD

**Discussion by HofmeisterAn** [testcontainers/testcontainers-dotnet#1275] [Ideas]
Hey 👋 I've noticed quite a few PRs that update the image version of modules. This is something we usually try to avoid because we don't want to introduce breaking changes or cause developers to run into issues when updating the Testcontainers version. Like with all dependencies, it’s [recommended](https://dotnet.testcontainers.org/api/best_practices/) to pin the version to avoid unexpected issues and to consciously update a version of a dependent service.

The idea behind having a default version was to provide a simple starting point for developers so they don’t have to immediately search through registries for the right and compatible image. Other Testcontainers language versions treat this differently, they require the image as a constructor argument from the start, e.g.:

```csharp
_ = new PostgreSqlBuilder("postgres:16.4-alpine")
```

Do you think we should introduce something similar in .NET as well?

**Comment by kiview** [comment]
You know, I am in favor of requiring being explicit about the version, since the implicit default version tends to lead to all kind of weird assumptions and expectations down the line 😉 

In an ideal world, the module could be even more explicit about supported version ranges, but even with this design, I believe it is a better DX if users need to be explicit with the used version for _most_ TC use cases.

**Reply by HofmeisterAn** [reply]
The more I see developers not following common best practices, the more I agree and support this change as well. 

In the matter of:

> so they don’t have to immediately search through registries for the right and compatible image.

We could add information (inkl. a link) about the expected repository to the XML documentation (class documentation).
